lib/geocoder/lookups/esri.rb

```diff
@@ -52,7 +52,14 @@ def token
         configuration[:token].to_s
       elsif configuration.api_key # generate a new token if we have credentials
         token_instance = Geocoder::EsriToken.generate_token(*configuration.api_key)
-        Geocoder.configure(:esri => {:token => token_instance})
```

Comment:
Thanks! That should work fine. However, it looks like the config state isn't being reset between tests so all the tests that don't use a token are failing now. I'll keep working on it.